src/js/stores/MesosStateStore.js

```diff
@@ -323,33 +255,34 @@ class MesosStateStore extends GetSetBaseStore {
     // Combine framework (if matching framework was found) and filtered
     // Marathon tasks. This will give you a list of framework tasks including
     // the scheduler tasks or a list of Marathon application tasks.
-    return frameworks.reduce(function(serviceTasks, framework) {
-      const {
-        tasks = [],
-        completed_tasks = [],
-        unreachable_tasks = [],
-        name
-      } = framework;
-      // Include tasks from framework match, if service is a Framework
-      if (serviceIsFramework && serviceFrameworkName === name) {
-        return serviceTasks
-          .concat(tasks, completed_tasks)
-          .map(task => assignSchedulerTaskField(task, schedulerTasks))
-          .map(task => flagSDKTask(task, service));
-      }
+    let serviceTasks = [];
+    if (serviceIsFramework) {
+      const framework = frameworks.find(function(framework) {
+        return framework.name === serviceFrameworkName;
+      });
 
-      // Filter marathon tasks by service name
-      if (name === "marathon") {
-        return tasks
-          .concat(completed_tasks, unreachable_tasks)
-          .filter(({ name }) => name === mesosTaskName)
-          .concat(serviceTasks)
-          .map(task => assignSchedulerTaskField(task, schedulerTasks))
-          .map(task => flagSDKTask(task, service));
+      if (framework) {
+        serviceTasks = tasks
+          .filter(task => task.framework_id === framework.id)
+          .map(task =>
+            MesosStateUtil.assignSchedulerTaskField(task, schedulerTasks)
+          )
+          .map(task => MesosStateUtil.flagSDKTask(task, service));
       }
+    }
+
+    const marathon = frameworks.find(framework => {
+      return framework.name === "marathon";
+    });
 
-      return serviceTasks;
-    }, []);
+    return tasks
+      .filter(task => task.framework_id === marathon.id)
+      .filter(({ name }) => name === mesosTaskName)
```

Comment:
Didn't you mention in another PR that you do not like to destruct in the parameter definition?

Reply:
There was a pr and we've agreed, and we actually even have an eslint rule for that, that we don't destruct in a method definition. Lambdas are ok.

Consider this example
```
class Tree  {
  bark({arg1, arg2, bla, bla}) { // <= This is restricted 
    ...
  }
}
```